cluster-autoscaler/cloudprovider/aws/aws_cloud_provider.go

```diff
@@ -143,6 +150,11 @@ func (aws *awsCloudProvider) HasInstance(node *apiv1.Node) (bool, error) {
 		return true, cloudprovider.ErrNotImplemented
 	}
 
+	// Skip SageMaker instances
+	if strings.HasPrefix(node.GetName(), "hyperpod") {
```

Comment:
I am trying to understand why do you install CAS on the cluster when the worker nodes naming convention doesn't work properly with CAS scheduler.